internal/registry/strategy_registry.go

```diff
@@ -0,0 +1,71 @@
+package registry
+
+import (
+	"sync"
+
+	"github.com/cloudflare/tf-migrate/internal/interfaces"
+)
+
+type StrategyRegistry struct {
+	strategies map[string]interfaces.ResourceTransformer
+	order      []string
+	mu         sync.RWMutex
+}
+
+func NewStrategyRegistry() *StrategyRegistry {
+	return &StrategyRegistry{
+		strategies: make(map[string]interfaces.ResourceTransformer),
+		order:      make([]string, 0),
+	}
+}
+
+func (r *StrategyRegistry) Register(strategy interfaces.ResourceTransformer) {
+	r.mu.Lock()
```

Comment:
How many things do we expect to be accessing the strategy registry at once? I'm guessing that this only exists because of the nondeterministic nature of the `init()` funcs we designed for.

None of this is needed if this is all coordinated statically.